enacts/onset/agronomy.py

```diff
@@ -73,38 +76,95 @@ def soil_plant_water_balance(
         total available water that represents the maximum water capacity of the soil.
     sminit : DataArray
         timeless soil moisture to initialize the loop with.
+    kc_params : DataArray
+        Crop Cultivar Kc parameters as a function of the inflexion points of the Kc curve,
+        expressed in consecutive daily time deltas originating from `planting_date`
+        as coordinate `kc_periods` (default `kc_params` =None in which case Kc is set to 1).
+    planting_date : DataArray
+        dates when planting (default `planting_date` =None -- not covered yet)
     time_dim : str, optional
         daily time dimension to run the balance against (default `time_dim` ="T").
         
     Returns
     -------
-    sm, drainage : Tuple of DataArray
-        daily soil moisture and drainage over the growing season.
+    sm, drainage, et_crop : Tuple of DataArray
+        daily soil moisture, drainage and crop evapotranspiration over the growing season.
         
     See Also
     --------
     soil_plant_water_step
     
+    Notes
+    -----
+    The daily evapotranspiration `et` can be scaled by a Crop Cultivar Kc
+    modelizing a crop needs in water according to the stage of its growth.
+    Kc is set to 1 outside of the growing period. i.e. before planting date
+    and after the last Kc curve inflexion point.
+    
+    Examples
+    --------
+    Example of kc_params:
+    
+    >>> kc_periods = pd.TimedeltaIndex([0, 45, 47, 45, 45], unit="D")
+    >>> kc_params = xr.DataArray(
+    >>>    data=[0.2, 0.4, 1.2, 1.2, 0.6], dims=["kc_periods"], coords=[kc_periods]
+    >>> )
+    <xarray.DataArray (kc_periods: 5)>
+    array([0.2, 0.4, 1.2, 1.2, 0.6])
+    Coordinates:
+        * kc_periods  (kc_periods) timedelta64[ns] 0 days 45 days ... 45 days 45 days
+    
+    Example of planting_date:
+    
+    >>> p_d = xr.DataArray(
+    >>>    pd.DatetimeIndex(data=["2000-05-02", "2000-05-13"]),
+    >>>    dims=["station"],
+    >>>    coords={"station": [0, 1]},
+    >>> )
+    <xarray.DataArray (station: 2)>
+    array(['2000-05-02T00:00:00.000000000', '2000-05-13T00:00:00.000000000'],
+        dtype='datetime64[ns]')
+    Coordinates:
+        * station        (station) int64 0 1
     """
     
     # First Step
     if np.size(et) == 1:
         et = xr.DataArray(et)
+    # Setting Kc
+    if kc_params is None:
+        kc = 1
+    else:
+        kc_inflex = kc_params.assign_coords(
+            kc_periods=kc_params["kc_periods"].cumsum(dim="kc_periods")
+        )
+        if planting_date is not None:
+            planted_since = peffective[time_dim][0].drop(time_dim) - planting_date
+            kc = kc_inflex.interp(
+                kc_periods=planted_since, kwargs={"fill_value": 1}
+            ).drop("kc_periods")
```

Comment:
If kc_params is not None, but planting date is None, then kc doesn't get defined? Then line 147 will throw an exception?

If failing is the right thing to do in that case, it would be more helpful to fail with an error message that says "if kc_params is provided then planting_date must also be provided", rather than the cryptic message about a reference to an undefined variable that will come from line 147 failing.

Reply:
I meant to cover that in the next PR with dynamic finding of planting_date (case planting date is None) so I didn't bother with an exception here. But can if you'd rather, even it if will be short lived.